src/main/java/week_by_week/week15/LinkedListCycleDetection.java

```diff
@@ -0,0 +1,89 @@
+package week_by_week.week15;
+
+
+import java.util.HashSet;
+import java.util.Set;
+
+
+public class LinkedListCycleDetection {
+
+
+    public ListNode detectCycleSet(ListNode head) {
+
+        Set<ListNode> set = new HashSet<>();
+
+        ListNode current = head;
+
+        while(current != null){
+            if(!set.add(current)){
+                return current;
```

Comment:
```suggestion  no need to change nothing i am just practising from intlj how to commint and bring suggestion or commit    ```
public ListNode detectCycleSet(ListNode head) {

        Set<ListNode> set = new HashSet<>();

        ListNode current = head;

        while(current != null){
            if(!set.add(current)){
                return current;

```  public ListNode detectCycleSet(ListNode head) {

        Set<ListNode> set = new HashSet<>();

        ListNode current = head;

        while(current != null){
            if(!set.add(current)){
                return current;```

Reply:
I got your point thanks Bobur